apps/cowswap-frontend/src/modules/hooksStore/pure/AddCustomHookForm/CustomDappLoader/index.tsx

```diff
@@ -54,23 +117,46 @@ export function ExternalDappLoader({
             url: input,
           })
         }
-      })
-      .catch((error) => {
-        if (!isRequestRelevant) return
-
-        console.error(error)
-        setManifestError('Can not fetch the manifest.json')
-      })
-      .finally(() => {
-        if (!isRequestRelevant) return
+      } catch (error) {
+        console.error('Hook dapp loading error:', error)
 
+        if (error.message?.includes('JSON')) {
+          setManifestError(
+            <>
+              Invalid manifest format
+              <br />
+              <small>Technical details: {error.message}</small>
+            </>,
+          )
+        } else if (error.name === 'AbortError') {
+          setManifestError('Request timed out. Please try again.')
```

Comment:
The text is repeated in `fetchWithTimeout`, would be great to move it in a const